### ISSUE TITLE
Question about the result

### ISSUE THREAD

**Issue by changhee1** [JiaRenChang/PSMNet#29]
Hello,
I executed 'submission.py' with 'pretrained_model_KITTI2015.tar'. but I got result below that isn't much like the result you posted. Did I something wrong?
![000000_10](https://user-images.githubusercontent.com/7540390/39416871-3b31ff2c-4c8b-11e8-839b-920f43569ae4.png)

**Comment by JiaRenChang** [comment]
@changhee1  Would you describe in more detail?

**Comment by changhee1** [comment]
@JiaRenChang I just downloaded your pretrained params and cloned your code, then executed this command `python submission.py --datapath [kitti_scene_flow_path]/testing/ --loadmodel pretrained_model_KITTI2015.tar` without any fine-tuning and I got '000000_10 ~ 000199_10.png' like above. thanks for replying.

**Comment by JiaRenChang** [comment]
Hi, @changhee1 
The problem may happen at saving disparity images.
You can try to replace code line 112 in the submission.py with "skimage.io.imsave(test_left_img[inx].split('/')[-1],(img).astype('uint8'))".

**Comment by JiaRenChang** [comment]
hi, @AdrienGhariani 
I met the similar problem on my coworker computer.
I just changed the save function, and it works fine.

**Comment by AdrienGhariani** [comment]
Using "skimage.io.imsave(test_left_img[inx].split('/')[-1],(img).astype('uint8'))" or just writing out the disparity array img as float with gdal did not help.
I am still getting the same exact results:
![000048_10](https://user-images.githubusercontent.com/5453315/39441375-8cfbe15c-4c62-11e8-9627-628db3f40f5d.png)
.

**Comment by JiaRenChang** [comment]
Oh..., I will check it soon.

**Comment by JiaRenChang** [comment]
Hi,
Would you test the training set and compare the output to ground truth?

**Comment by AdrienGhariani** [comment]
I tried an image from sceneflow dataset and the result is similar. Doesn't look like your results at all.
It feels like the weights from pretrained_model_KITTI2015.tar are not from the final epoch but one of the first epochs. Or something is corrupting them when we load them?

**Comment by JiaRenChang** [comment]
Well....I do not know, too.
I will try reproduce this bug using pytorch 0.4.0+.

**Comment by AdrienGhariani** [comment]
Going back to pytorch 0.3.1 and using volatile=True didn't help for me.
I am still getting the same exact result.
@changhee1, did you change anything else?

**Comment by AdrienGhariani** [comment]
I've found the issue!

Torchvision is the problem.
With torchvision 0.2.1 (default version) it doesn't work.
With torchvision 0.2.0 it works fine.

**Comment by JKozerawski** [comment]
Confirmed, downgrading to torchvision 0.2.0 solved to issue.

**Comment by JiaRenChang** [comment]
Thanks! I will add this solution to README.

**Comment by royanimal** [comment]
hello,I downgrad to torchvision 0.2.0 to solve the issue when I use uint16.But when I use uint8 and the torchvision is 0.2.0,I still got the wrong result.Why must use the uint16？Can you help me? Thank you very much.非常感谢。

**Comment by stalin18** [comment]
Yes, downgrading to torchvisoin 0.2.0 works for me too. @JiaRenChang , sorry if it is mentioned before, but the pretrained model which you have made available, is it the same final model the results of which you submitted to official KITTI test dataset evaluation? Thank you for help.

**Comment by JiaRenChang** [comment]
@stalin18  Hi, yes, the pretrained model we uploaded is converted from the final model we submit to KITTI.
@royanimal  KITTI disparity images were saved in uint16, so we saved in the same format. However, the disparity values all not exceeded 255, and they should be able to save in uint8.

**Comment by JiaRenChang** [comment]
@stalin18 It means that we rewrote the python file of the network, and renamed/spilted some modules. So we can't directly reload the network we trained before. We then converted network weights to renewed network.

**Comment by royanimal** [comment]
Hi,JiaRen
this code: 
 "skimage.io.imsave(save_dir, (img * 256).astype('uint16'))"
you save the img as uint16,and the img numpy array is float,and the astype finish the convert function.Can you tell me why you *256?
I use your code to compute the disparity,and I want to get the real disparity,like sgdm.
Thank you very much!

**Comment by JiaRenChang** [comment]
@royanimal 
*256 scales disparity to the range of uint16.
if you want real disparity, you can rewrite to skimage.io.imsave(save_dir, (img).astype('uint8')).

**Comment by xycim** [comment]
@AdrienGhariani how your final result pictures look like?

**Comment by AdrienGhariani** [comment]
Final results are similar to the published paper.
Though I save the disparity in float instead of uint8 or uint16.

**Comment by scholltan** [comment]
@JiaRenChang  @AdrienGhariani   do you have found the  place of the bug？  I downgrade torchvisoin to 0.2.0, with pytorch 0.4.0， it works for me. Is it the bug in torchvision or in this model?

**Comment by hurricane2018** [comment]
I test the model (pretrained_model_KITTI2015.tar), get the disparity result:
![image](https://user-images.githubusercontent.com/36166137/41263509-90a25f42-6db5-11e8-8af5-7f7da5b19770.png)
I looks like the result is right. However, when I assess the accuracy of the disparity. It has a very large error like this. 
![image](https://user-images.githubusercontent.com/36166137/41263547-be4a8050-6db5-11e8-9d72-d0edcc004084.png)

Could you guys help me ?
Thanks

**Comment by hurricane2018** [comment]
I test the model (pretrained_model_KITTI2015.tar), get the disparity result:

![image](https://user-images.githubusercontent.com/36166137/41263624-17b8b86e-6db6-11e8-8aed-666872fdb726.png)

It looks like the result is right. However, when I assess the accuracy of the disparity. It has a very large error, 46.5%  like this.

![image](https://user-images.githubusercontent.com/36166137/41263608-034e6dc4-6db6-11e8-9e1e-5be1e5328626.png)

Could you guys help me ?
Thanks

**Comment by EnJiang** [comment]
![image](https://user-images.githubusercontent.com/17062906/43180497-04aafb5c-900b-11e8-9f36-d340f58b1294.png)
![image](https://user-images.githubusercontent.com/17062906/43180518-17cee112-900b-11e8-8391-4e7edd555460.png)

![image](https://user-images.githubusercontent.com/17062906/43180512-1188b3b4-900b-11e8-8f37-04d73e56a874.png)
I used pytorch 0.4.0 and torchvision 0.2.0, still got a wrong result:
![image](https://user-images.githubusercontent.com/17062906/43180542-3ba4f11c-900b-11e8-8a2b-6bc3aea17c9e.png)
Any suggestion?

**Comment by LifeBeyondExpectations** [comment]
Do anyone have any solution for this problem ??? .. 
Still have no idea...
![000000_11](https://user-images.githubusercontent.com/32631453/43592714-286c12a4-96b1-11e8-93f1-252404757ea0.png)

**Comment by JiaRenChang** [comment]
@EnJiang 
Your result seems to be correct. The warning is just for upsample function, and you can skip the warning.

@LifeBeyondExpectations 
The output  of the network is a gray scale image (0~255).
We used skimage to save the image.
If you still got problem, you may try other image processing libraries such as PIL or opencv to save output image.
